### PR TITLE
Allow overwrite existing uploads

### LINKED ISSUE #21: Add ability to re-upload a file
When a partial submission contains a file, once it's submitted, it can't be updated. It should be possible to update the file upload with a newer or better upload.

### PR DESCRIPTION
**Pull request by lhalaa**
Resolves https://github.com/Firesphere/silverstripe-partial-userforms/issues/21

### CODE REVIEW

File: src/models/PartialFileFieldSubmission.php [hunk 1]
```diff
@@ -71,4 +71,12 @@ public function canDelete($member = null)
 
         return parent::canDelete($member);
     }
+
+    /**
+     * @return string
+     */
+    public function getFilename()
+    {
+        return $this->UploadedFileID ? $this->UploadedFile()->Name : '';
```
Comment (Firesphere) [inline src/models/PartialFileFieldSubmission.php]: Is this necessary? I thought the parent of PartialFileFieldSubmission already took care of this?

File: src/models/PartialFormSubmission.php [hunk 2]
```diff
@@ -233,4 +233,17 @@ public function generateKey($token)
...
+     * Get all partial fields for loading data into the form
+     *
+     * @return array
+     */
+    public function getFields()
+    {
+        $formFields = $this->PartialFields()->map('Name', 'Value')->toArray();
+        $fileFields = $this->PartialUploads()->map('Name', 'Filename')->toArray();
+
+        return $formFields + $fileFields;
```
Comment (Firesphere) [inline src/models/PartialFormSubmission.php]: Could you use array_merge here? I'm not a fan of using `+`, it works, but is unclear in its intention

**Review by Firesphere — CHANGES_REQUESTED** [review]

**Comment by Firesphere** [comment]
Could you add some tests? It's better to keep the testing quality up than letting it go

File: src/models/PartialFileFieldSubmission.php (hunk 1 above)
Comment (lhalaa) [reply src/models/PartialFileFieldSubmission.php]: Filename returns "Uploads/file.pdf", I just want the name.

**Comment by lhalaa** [comment]
> Could you add some tests? It's better to keep the testing quality up than letting it go

I won't have time to add all the tests, so my tests will be limited

**Comment by lhalaa** [comment]
Moved to https://github.com/Firesphere/silverstripe-partial-userforms/pull/32

**Pull request closed without merging**